### PR TITLE
[TechDraw] Process template upon filepath specification (#9990)

### PR DESCRIPTION
**Pull request by benj5378**
Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by benj5378** [comment]
Of what I tested it works fine, but it would be really sad to have introduced a bug into 21.0.

Also, we should probably write some more tests...

File: src/Mod/TechDraw/App/DrawSVGTemplate.cpp [hunk 1]
```diff
@@ -181,17 +181,24 @@ double DrawSVGTemplate::getHeight() const
...
+    Base::FileInfo tfi(Filepath.getValue());
+    if (!tfi.isReadable()) {
+        // If there is an old absolute template file set use a redirect
+        const std::string newPath = App::Application::getResourceDir() + "Mod/Drawing/Templates/" + tfi.fileName();
+        tfi.setFile(newPath);
+        // Try the redirect
+        if (tfi.isReadable()) {
+            Filepath.setValue(newPath.c_str());
+            return;
+        }
```
Comment (benj5378) [inline src/Mod/TechDraw/App/DrawSVGTemplate.cpp:200]: I haven't tested this

**Comment by WandererFan** [comment]
Once upon a time, template worked like the DrawingWB version and the svg(PageResult) was modified with the editable fields when they changed.  Now we only merge in the editable field values as required in processTemplate(), so we don't need to keep the modified svg.

The v0.21 tag is supposed to be created this weekend(?), so we can merge this next week maybe.

**Comment by WandererFan** [comment]
Add new page using default template (USLetter_Landscape_blank.svg) gives message:
20:21:12  DSVGT::checkFilepath(/home/wile/.cache/FreeCAD/Cache/FreeCAD_Doc_c90b97cc-6262-46ce-a7d8-fd2e7636952b_da39a3_1884894/USLetter_Landscape_blank.svg)

and again on changing FilePath property of Template to A4_Landscape_ISO7200TD.svg:
20:22:41  DSVGT::checkFilepath(/home/wile/.cache/FreeCAD/Cache/FreeCAD_Doc_c90b97cc-6262-46ce-a7d8-fd2e7636952b_da39a3_1884894/USLetter_Landscape_blank.svg)

clicking on a green edit handle and changing a field is not reflected on screen, but the EditableTexts property is updated.  Closing and reopening document updates the EditableTexts on screen.

Open pre-PR9991 document: 
Template.FilePath is empty.  Default template used instead of the template specified in the document.

**Comment by WandererFan** [comment]
This PR is too stale and template processing has changed in the interim.  Closing.

**Pull request closed without merging**